### DISCUSSION TITLE
Is there a way to control the zoom in the UI via a script?

### DISCUSSION THREAD

**Discussion by rmcginty** [spakin/SimpInkScr#135] [Q&A] [answered]
I may just not be understanding the relationship between the viewbox and the image, but is there a way to basically do a "Zoom Page" (or actually perform that command)?

**Comment by spakin** [comment, marked as answer]
I've been looking into this, and it appears the answer is "no".

An Inkscape SVG file represents the current zoom within a `sodipodi:namedview` element's `inkscape:zoom` attribute, as in the following:
<pre>
&lt;sodipodi:namedview
   id=&quot;base&quot;
   pagecolor=&quot;#ffffff&quot;
   bordercolor=&quot;#666666&quot;
   borderopacity=&quot;1.0&quot;
   inkscape:pageopacity=&quot;0.0&quot;
   inkscape:pageshadow=&quot;2&quot;
   <strong>inkscape:zoom=&quot;0.5&quot;</strong>
   inkscape:cx=&quot;512&quot;
   inkscape:cy=&quot;385&quot;
   inkscape:document-units=&quot;px&quot;
   inkscape:current-layer=&quot;layer1&quot;
   showgrid=&quot;false&quot;
   inkscape:window-width=&quot;1920&quot;
   inkscape:window-height=&quot;1029&quot;
   inkscape:window-x=&quot;0&quot;
   inkscape:window-y=&quot;0&quot;
   inkscape:window-maximized=&quot;1&quot;
   inkscape:document-rotation=&quot;0&quot;
   inkscape:showpageshadow=&quot;2&quot;
   inkscape:pagecheckerboard=&quot;0&quot;
   inkscape:deskcolor=&quot;#d1d1d1&quot;&gt;
  &lt;inkscape:grid
     type=&quot;xygrid&quot;
     id=&quot;grid2816&quot;
     empspacing=&quot;4&quot;
     visible=&quot;false&quot;
     enabled=&quot;true&quot;
     snapvisiblegridlinesonly=&quot;true&quot;
     spacingx=&quot;16&quot;
     spacingy=&quot;16&quot;
     originx=&quot;0&quot;
     originy=&quot;0&quot;
     units=&quot;px&quot; /&gt;
&lt;/sodipodi:namedview&gt;
</pre>

Although Simple Inkscape Scripting does not currently provide convenient access to named-view elements, a script can use lower-level APIs to access an SVG element's attributes.  For example, the current zoom level can be read like this:
```Python
nv = svg_root.namedview
zoom = float(nv.get('inkscape:zoom'))
```
The following code _ought_ to set the zoom, but in fact has no effect:
```Python
nv = svg_root.namedview
nv.set('inkscape:zoom', '1.2345')
```
(Even if that did work, you'd still have to compute the zoom level corresponding to "Zoom Page" in terms of the window and page sizes.  There's no automatic sizing that I know of.)

Interestingly, modifying many of the other `sodipodi:namedview` attributes does work.  My hypothesis is that modifications to window-related attributes are overwritten when Simple Inkscape Scripting terminates and Inkscape regains control.

**Comment by rmcginty** [comment]
Interesting.  I was thinking I was just missing something so I'm glad to find out there is more to it than I originally thought!  Thanks for the quick response!
